cql3/prepare_context.cc

```diff
@@ -30,11 +30,11 @@ std::vector<uint16_t> prepare_context::get_partition_key_bind_indexes(const sche
     auto count = schema.partition_key_columns().size();
     std::vector<uint16_t> partition_key_positions(count, uint16_t(0));
     std::vector<bool> set(count, false);
-    for (size_t i = 0; i < _target_columns.size(); i++) {
-        auto& target_column = _target_columns[i];
-        const auto* cdef = target_column ? schema.get_column_definition(target_column->name->name()) : nullptr;
+
+    for (auto&& [bind_index, target_spec] : _targets) {
+        const auto* cdef = target_spec ? schema.get_column_definition(target_spec->name->name()) : nullptr;
         if (cdef && cdef->is_partition_key()) {
-            partition_key_positions[cdef->position()] = i;
+            partition_key_positions[cdef->position()] = bind_index;
```

Comment:
If I understand correctly, when multiple partition key columns are involved (a composite partition key), this will list them in the order they were mentioned in the request, not in the order of the partition key columns' definition. This sounds to me like a problem - the driver might look at the bound values in the wrong order when computing the token.

Reply:
Thanks